core/capabilities/remote/utils_test.go

```diff
@@ -90,29 +89,32 @@ func TestToPeerID(t *testing.T) {
 }
 
 func TestDefaultModeAggregator_Aggregate(t *testing.T) {
-	capResponse1 := marshalCapabilityResponse(t, triggerEvent1, nil)
-	capResponse2 := marshalCapabilityResponse(t, triggerEvent2, nil)
-
-	agg := remote.NewDefaultModeAggregator(2)
-	_, err := agg.Aggregate("", [][]byte{capResponse1})
-	require.Error(t, err)
-
-	_, err = agg.Aggregate("", [][]byte{capResponse1, capResponse2})
-	require.Error(t, err)
-
-	res, err := agg.Aggregate("", [][]byte{capResponse1, capResponse2, capResponse1})
+	val, err := values.Wrap(triggerEvent1)
+	require.NoError(t, err)
+	capResponse1 := commoncap.CapabilityResponse{
+		Value: val,
+		Err:   nil,
+	}
+	marshaled1, err := pb.MarshalCapabilityResponse(capResponse1)
 	require.NoError(t, err)
-	require.True(t, bytes.Equal(res, capResponse1))
-}
 
-func marshalCapabilityResponse(t *testing.T, capValue any, capError error) []byte {
-	val, err := values.Wrap(capValue)
+	val2, err := values.Wrap(triggerEvent2)
 	require.NoError(t, err)
-	capResponse := commoncap.CapabilityResponse{
-		Value: val,
-		Err:   capError,
+	capResponse2 := commoncap.CapabilityResponse{
+		Value: val2,
+		Err:   nil,
 	}
-	marshaled, err := pb.MarshalCapabilityResponse(capResponse)
+	marshaled2, err := pb.MarshalCapabilityResponse(capResponse2)
+	require.NoError(t, err)
+
+	agg := remote.NewDefaultModeAggregator(2)
+	_, err = agg.Aggregate("", [][]byte{marshaled1})
+	require.Error(t, err, "failed to aggregate responses, err: not enough identical responses found")
```

Comment:
prefer to make the error a type or var and check with assert/require `ErrorIs` rather than string matching.

this approach will also play nicely with wrapping errors via %w

Reply:
hmm I'm not sure how to do that and still support wrapped inner errors - do you have an example somewhere?
I also realized that the expected string matching here doesn't actually work that way so I'm removing it.